scenes/Player/Player.gd

```diff
@@ -13,7 +12,11 @@ onready var flame_exhaust: Node2D = $Ship_Exhaust
 func _ready() -> void:
 	flame_exhaust.hide()
 
+
 func _physics_process(delta: float) -> void:
+	if PlayerState.hull_health <= 0: 
```

Comment:
Maybe move this into `_get_collission_damage` or at least below the call to `_get_collission_damage` so that it gets applied right when the health goes to zero.

Reply:
I think physics process runs like 60 times in a second? I think we only want to check for death once but putting it at the end of the physics process works for me.